cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -551,6 +552,28 @@ private Optional<EmittedRecords> tryRelationalTransform(
     }
 
     if (plugin instanceof RelationalTransform) {
+      // If there are multiple inputs to a "LinearRelationalTransform" implemented stage, then combine all the inputs
+      // into 1 dataset, as LinearRelationalTransform expects only 1 input.
+      if ((plugin instanceof LinearRelationalTransform) && inputDataCollections.size() > 1) {
+        Iterator<String> keys = inputDataCollections.keySet().iterator();
+        String firstKey = keys.next();
+        Schema firstSchema = stageSpec.getInputSchemas().get(firstKey);
+        SparkCollection<Object> first = inputDataCollections.get(firstKey);
+
+        while (keys.hasNext()) {
+          String currKey = keys.next();
+          //Validate schema is Same for all
+          if (firstSchema.equals(stageSpec.getInputSchemas().get(currKey))) {
+            first = first.union(inputDataCollections.get(currKey));
+          } else {
+            //If there is schema mismatch, return empty option (handle via usual cdap flow)
```

Comment:
We should probably at least warn here if not add a metric. This way we can answer "why the pushdown does not happen"